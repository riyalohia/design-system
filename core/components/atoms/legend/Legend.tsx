import * as React from 'react';
import { IconType } from '@/components/atoms/icon';
import Text, { Appearance as LabelAppearance, } from '@/components/atoms/text';
import classNames from 'classnames';

export interface LegendProps {
  /**
   * Name of icon to be rendered inside `Legend`
   */
  icon?: string;
  /**
   * Describes label of the `Legend`
   */
  label: string;
  /**
   * Color of Icon
   */
  iconAppearance?: string;
  /**
   * Color of label
   */
  labelAppearance?: LabelAppearance;
  /**
   * Type of Icon
   */
  iconType?: IconType;
  /**
   * Size of Icon
   */
  iconSize?: number;
  /**
   * Denotes weight of `text`
   */
  labelWeight?: 'strong' | 'medium';
  /**
   * Adds custom CSS to `Legend`
   */
  style?: React.CSSProperties;
  onClick?: (e: React.MouseEvent<HTMLDivElement>) => void;
  onMouseEnter?: (e: React.MouseEvent<HTMLDivElement>) => void;
  onMouseLeave?: (e: React.MouseEvent<HTMLDivElement>) => void;
}

const Legend = (props: LegendProps) => {
  const {
    icon = 'fiber_manual_record',
    iconAppearance = 'inverse',
    iconType = 'filled',
    iconSize = 14,
    labelAppearance,
    label,
    labelWeight,
    onMouseEnter,
    onMouseLeave,
    onClick,
    style,
  } = props;

  const LegendClass = classNames({
    ['Legend']: true,
  });

  const styles = {
    color: `var(--${iconAppearance})`,
    fontSize: `${iconSize}px`,
    width: `${iconSize}px`,
    marginRight: 'var(--spacing)',
  };

  return (
    <div
      className={LegendClass}
      onClick={e => onClick && onClick(e)}
      onMouseEnter={e => onMouseEnter && onMouseEnter(e)}
      onMouseLeave={e => onMouseLeave && onMouseLeave(e)}
      style={style}
    >
      <i
        className={'material-icons'}
        style={styles}
      >
        {`${icon}_${iconType}`}
      </i>
      <Text
        appearance={labelAppearance}
        weight={labelWeight}
      >
        {label}
      </Text>
    </div>
  );
};

export default Legend;
