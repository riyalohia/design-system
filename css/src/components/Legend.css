.Legend {
    display: flex;
    flex-direction: row;
    align-items: center;
    width: fit-content;
}

.Legend:hover {
    cursor: pointer;
}