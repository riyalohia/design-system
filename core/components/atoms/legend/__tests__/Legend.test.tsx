import * as React from 'react';
import { shallow } from 'enzyme';
import Legend, { LegendProps as IProps } from '../Legend';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const weight = ['strong', 'medium'];
const labelAppearance = ['default', 'white', 'destructive', 'subtle', 'disabled'];
const iconAppearance = [
  'primary',
  'secondary',
  'success',
  'alert',
  'warning',
  'accent1',
  'accent2',
  'accent3',
  'accent4',
  'inverse',
];
const label = 'Legend';

describe('Legend component', () => {
  const mapper = {
    labelWeight: valueHelper(weight, { required: true, iterate: true }),
    label: valueHelper(label, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Legend {...attr} />
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Legend component', () => {
  const mapper = {
    labelAppearance: valueHelper(labelAppearance, { required: true, iterate: true }),
    label: valueHelper(label, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Legend {...attr} />
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Legend component', () => {
  const mapper = {
    iconAppearance: valueHelper(iconAppearance, { required: true, iterate: true }),
    label: valueHelper(label, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Legend {...attr} />
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});
