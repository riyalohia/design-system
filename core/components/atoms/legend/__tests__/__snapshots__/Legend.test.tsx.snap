// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component 
  iconAppearance: "accent1", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--accent1)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text>
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  iconAppearance: "accent2", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--accent2)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text>
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  iconAppearance: "accent3", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--accent3)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text>
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  iconAppearance: "accent4", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--accent4)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text>
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  iconAppearance: "alert", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--alert)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text>
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  iconAppearance: "inverse", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--inverse)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text>
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  iconAppearance: "primary", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--primary)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text>
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  iconAppearance: "secondary", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--secondary)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text>
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  iconAppearance: "success", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--success)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text>
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  iconAppearance: "warning", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--warning)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text>
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  labelAppearance: "default", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--inverse)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text
    appearance="default"
  >
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  labelAppearance: "destructive", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--inverse)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text
    appearance="destructive"
  >
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  labelAppearance: "disabled", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--inverse)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text
    appearance="disabled"
  >
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  labelAppearance: "subtle", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--inverse)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text
    appearance="subtle"
  >
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  labelAppearance: "white", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--inverse)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text
    appearance="white"
  >
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  labelWeight: "medium", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--inverse)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text
    weight="medium"
  >
    Legend
  </Text>
</div>
`;

exports[`Avatar component 
  labelWeight: "strong", label: "Legend"
 1`] = `
<div
  className="Legend"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="material-icons"
    style={
      Object {
        "color": "var(--inverse)",
        "fontSize": "14px",
        "marginRight": "var(--spacing)",
        "width": "14px",
      }
    }
  >
    fiber_manual_record_filled
  </i>
  <Text
    weight="strong"
  >
    Legend
  </Text>
</div>
`;
